src/Client/Connection.cpp

```diff
@@ -819,6 +824,14 @@ void Connection::sendQuery(
 
     query_id = query_id_;
 
+    /// Avoid reusing connections that had been left in the intermediate state
+    /// (i.e. not all packets had been sent).
+    bool completed = false;
+    SCOPE_EXIT_MEMORY_SAFE({
```

Comment:
What are considerations behind `SCOPE_EXIT_MEMORY_SAFE` usage here? Shouldn't we just put the code in try ... catch block?